src/module.indexer/model/poolswap.aggregation.ts

```diff
@@ -0,0 +1,110 @@
+import { Injectable } from '@nestjs/common'
+import { PoolSwapAggregation, PoolSwapAggregationMapper } from '@src/module.model/poolswap.aggregation'
+import { Indexer, RawBlock } from './_abstract'
+import { OP_DEFI_TX } from '@defichain/jellyfish-transaction/dist/script/defi'
+import { toOPCodes } from '@defichain/jellyfish-transaction/dist/script/_buffer'
+import { SmartBuffer } from 'smart-buffer'
+import BigNumber from 'bignumber.js'
+
+@Injectable()
+export class PoolSwapAggregationIndexer extends Indexer {
+  constructor (
+    private readonly mapper: PoolSwapAggregationMapper
+  ) {
+    super()
+  }
+
+  async index (block: RawBlock): Promise<void> {
+    for (const txn of block.tx) {
+      for (const vout of txn.vout) {
+        // grab poolswap tx only
+        if (!vout.scriptPubKey.asm.startsWith('OP_RETURN 4466547873')) {
+          continue
+        }
+        const stack: any = toOPCodes(
+          SmartBuffer.fromBuffer(Buffer.from(vout.scriptPubKey.hex, 'hex'))
+        )
+
+        const data = (stack[1] as OP_DEFI_TX).tx.data
+        const poolId = constructPoolId(data.fromTokenId, data.toTokenId)
+        const bucketId = roundMinutes(block.time)
+        const id = constructId(poolId, bucketId)
+
+        let aggregation = await this.mapper.get(id)
+        if (aggregation === undefined) {
+          console.log('aggregation undefined: ', aggregation)
+          aggregation = PoolSwapAggregationIndexer.newPoolSwapAggregation(poolId, bucketId)
+        }
+
+        aggregation.total = new BigNumber(aggregation.total).plus(data.fromAmount)
+        aggregation.count += 1
+
+        console.log('aggregation: ', aggregation)
+
+        await this.mapper.put(aggregation)
+
+        // manually add 1 here and test query
+        const testagg = PoolSwapAggregationIndexer.newPoolSwapAggregation(poolId, '2020-08-31T19:20')
```

Comment:
```suggestion
        const testagg = PoolSwapAggregationIndexer.newPoolSwapAggregation(poolId, bucketId)
```

you are using a different id here than the `const id` hence you get nothing

Reply:
no no.. here is to ensure the indexed data is stored against the test as test got undefined by getting indxed data